src/microbe_stage/Microbe.Interior.cs

```diff
@@ -378,6 +378,34 @@ public void SetInitialCompounds()
         {
             Compounds.AddCompound(entry.Key, entry.Value);
         }
+
+        var setting = GameWorld.WorldSettings;
+
+        // Fraction of day that is night
+        var nightTimeFraction = 1 - setting.DaytimeFraction;
+
+        // Shifts fraction so the day ends exactly 1 (ignoring padding).
```

Comment:
I think this comment is insufficient to explain the complexity that is the next two lines, which I also think should be put in as a method in DayNightCycle to concentrate the tricky calculations there.